openeogeotrellis/job_tracker.py

```diff
@@ -181,7 +181,8 @@ def update_statuses(self) -> None:
                                         async_task.schedule_delete_batch_process_dependency_sources(
                                             job_id, user_id, dependency_sources)
 
-                                    registry.mark_done(job_id, user_id)
+                                    # TODO: this `set_status` is just here to trigger the "mark_done from status" logic. Can this be automated further?
+                                    registry.set_status(job_id, user_id, new_status)
```

Comment:
Same here, there's a status change a few lines earlier as well.

Reply:
as mentioned earlier, there is a `remove_dependencies` following on the `patch`, so changing the zookeeper path of a job indirectly from `patch` seemed like risky to include at this point.

but certainly to reconsider as a next step